File: CS-Examples/11_Formatting/SetRowColorByConditionalFormat.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;
using Spire.Xls;
using Spire.Xls.Core.Spreadsheet;
using Spire.Xls.Core.Spreadsheet.Collections;
using Spire.Xls.Core;

namespace SetRowColorByConditionalFormat
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }

		private void btnRun_Click(object sender, System.EventArgs e)
		{
            //Create a workbook.
			Workbook workbook = new Workbook();

            //Load the file from disk.
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Xls_4.xlsx");

            //Get the first worksheet.
			Worksheet sheet = workbook.Worksheets[0];

            //Select the range that you want to format.
            CellRange dataRange = sheet.AllocatedRange;

            //Set conditional formatting.
            XlsConditionalFormats xcfs = sheet.ConditionalFormats.Add();
            xcfs.AddRange(dataRange);
            IConditionalFormat format1 = xcfs.AddCondition();
            //Determines the cells to format.
            format1.FirstFormula = "=MOD(ROW(),2)=0";
            //Set conditional formatting type
            format1.FormatType = ConditionalFormatType.Formula;
            //Set the color.
            format1.BackColor = Color.LightSeaGreen;

            //Set the backcolor of the odd rows as Yellow.
            XlsConditionalFormats xcfs1 = sheet.ConditionalFormats.Add();
            xcfs1.AddRange(dataRange);
            IConditionalFormat format2 = xcfs.AddCondition(); 
            format2.FirstFormula = "=MOD(ROW(),2)=1";
            format2.FormatType = ConditionalFormatType.Formula;
            format2.BackColor = Color.Yellow;

            // Specify the name for the resulting Excel file
            String result = "Result-SetRowColorWithConditionalFormatting.xlsx";

            // Save the workbook to the specified file in Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launch the MS Excel file.
            ExcelDocViewer(result);
		}

        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}

File: CS-Examples/11_Formatting/SetRowColorByConditionalFormat.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports Spire.Xls
Imports Spire.Xls.Core.Spreadsheet
Imports Spire.Xls.Core.Spreadsheet.Collections
Imports Spire.Xls.Core

Namespace SetRowColorByConditionalFormat
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new instance of Workbook
            Dim workbook As New Workbook()

            ' Load an existing Excel file
            workbook.LoadFromFile("..\..\..\..\..\..\Data\Template_Xls_4.xlsx")

            ' Get the first worksheet in the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Get the range of cells that contain data in the worksheet
            Dim dataRange As CellRange = sheet.AllocatedRange

            ' Add a new conditional formats collection to the worksheet
            Dim xcfs As XlsConditionalFormats = sheet.ConditionalFormats.Add()
            ' Add the data range to the conditional formats collection
            xcfs.AddRange(dataRange)
            ' Add a new conditional format to the collection
            Dim format1 As IConditionalFormat = xcfs.AddCondition()
            ' Set the formula for the conditional format (highlight even rows)
            format1.FirstFormula = "=MOD(ROW(),2)=0"
            ' Set the format type to Formula
            format1.FormatType = ConditionalFormatType.Formula
            ' Set the background color to light sea green
            format1.BackColor = Color.LightSeaGreen

            ' Add another conditional formats collection to the worksheet
            Dim xcfs1 As XlsConditionalFormats = sheet.ConditionalFormats.Add()
            ' Add the data range to the second conditional formats collection
            xcfs1.AddRange(dataRange)
            ' Add a new conditional format to the second collection
            Dim format2 As IConditionalFormat = xcfs.AddCondition()
            ' Set the formula for the conditional format (highlight odd rows)
            format2.FirstFormula = "=MOD(ROW(),2)=1"
            ' Set the format type to Formula
            format2.FormatType = ConditionalFormatType.Formula
            ' Set the background color to yellow
            format2.BackColor = Color.Yellow

            ' Save the workbook to a new file with the name "Result-SetRowColorWithConditionalFormatting.xlsx"
            Dim result As String = "Result-SetRowColorWithConditionalFormatting.xlsx"
            workbook.SaveToFile(result, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()
            'Launch the MS Excel file.
            ExcelDocViewer(result)
		End Sub

		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace
